packages/react-lowcode/src/codegen/app.ts

```diff
@@ -27,46 +31,89 @@ export function generatePages(inputSourceCode: string, io: CodeRW & CodeDir, opt
+            '',
+            ts.ScriptTarget.ESNext,
+            true,
+            ts.ScriptKind.TSX
+        )
+
+        const pageSourceCode = printer.printList(ts.ListFormat.MultiLine, factory.createNodeArray([...page!.imports, page!.functionDeclaration]), sourceFile)
+        
+        console.log(`table for ${typeName} was generated: ${pageSourceCode}`)
+        io.writeFile(filePath, pageSourceCode)
+    }
+
+    function generateListPage(io: CodeRW, 
+        entity: Entity, 
+        typeName:string,
+        pageListTemplateSource: string,
+        listPageFilePath: string) {
+        const templateResolver = new TemplateResolver(entity);
+        const listPage = templateResolver.generateListPage(pageListTemplateSource);
+
+        if(listPage) {
+            const listPageSourceFile = ts.createSourceFile(
+                listPageFilePath,
+                listPage,
                 ts.ScriptTarget.ESNext,
                 true,
                 ts.ScriptKind.TSX
             )
 
-            const pageSourceCode = printer.printList(ts.ListFormat.MultiLine, factory.createNodeArray([...page!.imports, page!.functionDeclaration]), sourceFile)
-            
-            console.log(`table for ${typeName} was generated: ${pageSourceCode}`)
-            io.writeFile(filePath, pageSourceCode)
+            const generatedSourceCode = printer.printFile(listPageSourceFile);
 
-            //generate list wrapper
-            const templateResolver = new TemplateResolver(entity);
-            const listWrapper = templateResolver.generateListPage(options.pageListTemplate);
+            console.log(`page for ${typeName} list was generated: ${generatedSourceCode}`)
 
-            if(listWrapper) {
-                const listWrapperFilePath = `src/components/${typeName}Page.tsx`
-                const sourceFileWrapperSourceFile = ts.createSourceFile(
-                    listWrapperFilePath,
-                    listWrapper,
-                    ts.ScriptTarget.ESNext,
-                    true,
-                    ts.ScriptKind.TSX
-                )
-    
-                // TODO:PC: Need print here? or only: io.writeFile(listWrapperFilePath, listWrapper)
-                const wrapperPageSourceCode = printer.printFile(sourceFileWrapperSourceFile);
+            io.writeFile(listPageFilePath, generatedSourceCode)
+        }
+    }
 
-                console.log(`page for ${typeName} table was generated: ${wrapperPageSourceCode}`)
+    function addNewListRoute(io:CodeRW,
+        entityName: string, 
+        componentFilePath: string){
+            //for the beginning path to route json definition will be harcoded
+            const routeDefinitionFilePath = 'src/routes.tsx'
 
-                io.writeFile(listWrapperFilePath, wrapperPageSourceCode)
-            }
-        }
-    })
+            generateRoute(
+                { routeFilePath: routeDefinitionFilePath, 
+                  componentName: 'EntityHelper.getEntityName()', 
```

Comment:
What is this constant 'EntityHelper.getEntityName()'`?
Don't we needt`componentRouteUri`?